### DISCUSSION TITLE
'argument list too long' error

### DISCUSSION THREAD

**Discussion by cwrx777** [k3s-io/k3s#9713] [Q&A]
Hi,

k3s version v1.29.1+k3s2 (57482a1c)
go version go1.21.6
OS: RHEL 8.9

I installed k3s using the airgap image, and all binaries linked to k3s cannot be run using sudo, i.e.:

![image](https://github.com/k3s-io/k3s/assets/3224215/4ef81a53-a455-4d95-b9b4-5575b3c6da77)

```shell
$ sudo ctr version
FATA[0000] exec /usr/local/bin/ctr failed: argument list too long

$ sudo kubectl version
FATA[0000] exec /usr/local/bin/kubectl failed: argument list too long

$ sudo crictl version
FATA[0000] exec /usr/local/bin/crictl failed: argument list too long
```
no issue if they r run without sudo, but there is permission error on containerd.sock
```
$ ctr version
Client:
  Version:  v1.7.11-k3s2
  Revision:
  Go version: go1.21.6

ctr: failed to dial "/run/k3s/containerd/containerd.sock": connection error: desc = "transport: error while dialing: dial unix /run/k3s/containerd/containerd.sock: connect: permission denied"
```

`k3s` is OK, 
```shell
$ sudo k3s --version
k3s version v1.29.1+k3s2 (57482a1c)
go version go1.21.6

$ k3s --version
k3s version v1.29.1+k3s2 (57482a1c)
go version go1.21.6

```

FYI, the server is run using custom data-dir. 

Update:
ctr was able to run if I run using k3s and specify the data-dir, e.g.
```shell
$ sudo k3s --data-dir /app/k3s/data/ ctr version
Client:
  Version:  v1.7.11-k3s2
  Revision:
  Go version: go1.21.6

Server:
  Version:  v1.7.11-k3s2
  Revision:
  UUID: 8712c2b3-f28d-4f12-b304-deef27eb865f

$ sudo kubectl --data-dir /app/k3s/data/ ctr version
Client Version: v1.29.1+k3s2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.1+k3s2

$ sudo kubectl --data-dir /app/k3s/data/ crictl version
Version:  0.1.0
RuntimeName:  containerd
RuntimeVersion:  v1.7.11-k3s2
RuntimeApiVersion:  v1

```

**Comment by brandond** [comment]
I can't reproduce this. Can you strace the sudo command to see what all is getting passed to it? `sudo` isn't even getting as far as running any of our commands, it is failing first - so I'm not sure how this is something on our side.

**Reply by cwrx777** [reply]
Hi,

I managed to run it if I specified `--data-dir`.

**Reply by brandond** [reply]
I'm still curious what was going on.

**Reply by cwrx777** [reply]
Hi,

For your curiosity, please see the `strace` output in the gist:

[`sudo strace -f sudo ctr version`](https://gist.github.com/cwrx777/0e801adebf436ebdfab66ccd7b16e364)
[`sudo strace -f sudo k3s --data-dir /app/k3s/data ctr version`](https://gist.github.com/cwrx777/d2eb7baf5f6e8c885a7ed50658da6e3d)

**Comment by brandond** [comment]
I don't see an actual failure message there anywhere, but it looks like something is being sent to syslog?
`sendto(10, "<86>Mar 12 09:06:14 sudo: pam_un"..., 101, MSG_NOSIGNAL, NULL, 0) = 101`
